tree_test.go

```diff
@@ -1391,13 +1466,20 @@ func TestManipulateChildren(t *testing.T) {
 }
 
 func TestLeafNodeInsert(t *testing.T) {
+	t.Parallel()
+
+	keyTest := make([]byte, len(ffx32KeyTest))
+	copy(keyTest, ffx32KeyTest)
 	values := make([][]byte, NodeWidth)
 	valIdx := 42
 	values[valIdx] = testValue
-	ln, _ := NewLeafNode(ffx32KeyTest[:StemSize], values)
+	ln, err := NewLeafNode(keyTest[:StemSize], values)
+	if err != nil {
+		t.Fatalf("failed to create leaf node: %v", err)
+	}
```

Comment:
Here was the bug that became apparent when paralellizing (quite dangerous!).

The bug was in the deleted line: `append(ffx32KeyTest[:StemSize], byte(valIdx))`
That is literally appending bytes to `ffx32Keytest`, which is global. Now just make a copy in the test and do whatever we want.